### PR TITLE
Add power assertion output to asserts with explicit message

### LINKED ISSUE #922: assert with message disables power assertion output
With a feature method like this

```
def one() {
	expect:
	("one".capitalize() * 3).toUpperCase() == "OneOneOne"
}
```

we get an error message like that:
```
Condition not satisfied:

("one".capitalize() * 3).toUpperCase() == "OneOneOne"
       |            |    |             |
       One          |    ONEONEONE     false
                    OneOneOne          6 differences (33% similarity)
                                       O(NE)O(NE)O(NE)
                                       O(ne)O(ne)O(ne)

Expected :OneOneOne

Actual   :ONEONEONE
```

The same is true if we explicitly use `assert` at the beginning of the condition. So far, so good.

Now this is what happens when using `assert` with an additional error message:

```
def one() {
	expect:
	assert ("one".capitalize() * 3).toUpperCase() == "OneOneOne", "My custom message"
}
```

The very helpful power assert output vanishes.
```
Condition not satisfied:

("one".capitalize() * 3).toUpperCase() == "OneOneOne"

My custom message
```

What I would like to see instead is the custom message prepended to the power assert output like this:

```
My custom message

Condition not satisfied:

("one".capitalize() * 3).toUpperCase() == "OneOneOne"
       |            |    |             |
       One          |    ONEONEONE     false
                    OneOneOne          6 differences (33% similarity)
                                       O(NE)O(NE)O(NE)
                                       O(ne)O(ne)O(ne)

Expected :OneOneOne

Actual   :ONEONEONE
```

Would it be possible to add a corresponding AST transformation for that kind of syntax?

### PR DESCRIPTION
**Pull request by leonard84**
fixes #922

---
This change is [Reviewable](https://reviewable.io/reviews/spockframework/spock/928)

### CODE REVIEW

**Comment by leonard84** [comment]
@kriegaex what do you think?

**Comment by kriegaex** [comment]
@leonard84, having read the commit diffs on my iPad but not having given it a spin on a Spock-enabled machine using a snapshot build, today I can only say: It looks exactly like what I wanted, being a simple (as in non-configurable) implementation. As the new default behaviour is what I need, I can easily live without it being configurable. Suppressing the output manually if needed, using a helper methld or similar, should not be a big deal for anyone disliking the power-assert output.

Thank you so much! I will provide more feedback after a live test, probably as early as tomorrow.

**Comment by kriegaex** [comment]
I built your PR branch and ran my own test against it and it looks fine. The custom message is below the power assert output instead of above as I would expect, but maybe this is in line with what Groovy usually does, I did not check. Anyway, the message is there and that is a big step ahead. Thank you again. :-)

File: docs/release_notes.adoc [hunk 1]
```diff
@@ -4,6 +4,36 @@ include::include.adoc[]
 
 == 1.3 (tbd)
 
+Potential Breaking change: Assertions with explicit messages now include power assertions output.
```
Comment (marcphilipp) [inline docs/release_notes.adoc]: Nit: `breaking` and `assertions` should be lowercase.

**Pull request merged**